table/block_based/block_test.cc

```diff
@@ -43,6 +43,7 @@ std::string GenerateKey(int primary_key, int secondary_key, int padding_size,
   if (padding_size) {
     k += RandomString(rnd, padding_size);
   }
+  AppendInternalKeyFooter(&k, 0 /* seqno */, kTypeValue);
```

Comment:
Purpose? (Extra tricky keys?)

Reply:
It's because `DataBlockIter` no longer offers its user a choice of what comparator to use. Previously this file called `reader.NewDataIterator(options.comparator, options.comparator, ...` to get a `DataBlockIter` that used `options.comparator` for all its comparisons. However, now `DataBlockIter` offers no such choice and will use an `InternalKeyComparator(options.comparator)` for all its comparisons, so our test cases now need to forge data blocks containing internal keys.

The way the new code selects internal comparator is as follows:
1. `DataBlockIter::Initialize()` calls `raw_key_.SetIsUserKey(false);`
2. `BlockIter::CompareCurrentKey()` in that case uses `icmp_`

Reply:
Renaming the function GenerateInternalKey I think would suffice. (Unqualified 'key' means user key.)